homeassistant/components/envisalink/switch.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Envisalink zone bypass switches."""
+import logging
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import (
+    CONF_ZONENAME,
+    DATA_EVL,
+    SIGNAL_ZONE_BYPASS_UPDATE,
+    ZONE_SCHEMA,
+    EnvisalinkDevice,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Envisalink switch devices."""
+    configured_zones = discovery_info["zones"]
+
+    devices = []
+    for zone_num in configured_zones:
+        device_config_data = ZONE_SCHEMA(configured_zones[zone_num])
+        device = EnvisalinkSwitch(
+            hass,
+            zone_num,
+            device_config_data[CONF_ZONENAME] + "_bypass",
+            hass.data[DATA_EVL].alarm_state["zone"][zone_num],
+            hass.data[DATA_EVL],
+        )
+        devices.append(device)
+
+    async_add_entities(devices)
+
+
+class EnvisalinkSwitch(EnvisalinkDevice, SwitchEntity):
+    """Representation of an Envisalink switch."""
+
+    def __init__(self, hass, zone_number, zone_name, info, controller):
+        """Initialize the switch."""
+        self._zone_number = zone_number
+
+        _LOGGER.debug("Setting up zone_bypass switch: %s", zone_name)
+        super().__init__(zone_name, info, controller)
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        async_dispatcher_connect(
```

Comment:
Wrap this in `self.async_on_remove`

Reply:
@bdraco I see the pattern you suggested used in many components and it is used in some other places in this `envisalink` component too.  I'm happy to make this change but I'm new to HA development so can you help me understand what this is doing?  From what I can tell, wrapping the call in `async_on_remove` will change the behaviour of this function such that `async_dispatcher_connect` will no longer be called here (when the entity is being added?) and instead be called when the entity is removed.  Thanks!

Reply:
When the entity is removed from Home Assistant, it will remove the dispatcher.

Reply:
Ah, I misread the function signatures.  Makes sense, thanks!